flaml/automl/model.py

```diff
@@ -2031,7 +2040,7 @@ def search_space(cls, data_size, pred_horizon, **params):
         space.update(
             {
                 "optimize_for_horizon": {
-                    "domain": tune.choice([True, False]),
+                    "domain": tune.choice([False]),

```

Comment:
This change modifies the original search range.

Reply:
This is part of this change to fix a test

```
tree c6a5dd1a0e711c590b51c3e9c697eafd9fa4f7ac
parent 9df5200ddab810f72ab82800450c602c1239791a
author EgorKraevTransferwise <egor.kraev@transferwise.com> Tue Feb 7 09:58:42 2023 +0000
committer EgorKraevTransferwise <egor.kraev@transferwise.com> Tue Feb 7 09:58:42 2023 +0000

Hotwire TS_Sklearn wrapper to fix test fail by setting optimizer_for_horizon == False
```

I merged it in to get the tests passing for this PR

Reply:
yea, we can disable it for now... seemed to work fine before but I'll take a look into it after this PR.

Reply:
> yea, we can disable it for now... seemed to work fine before but I'll take a look into it after this PR.

Allowing this change here would decrease the code coverage and make buggy code stay in the repo without being covered by test. The right solution is to either remove or fix the buggy part. For example, if for some reason we don't support `optimize_for_horizon`=True, we should either fix it or remove the code corresponding to that case, and document the reason it doesn't work.

Reply:
We could change the test prediction horizon to have the same length as the validation horizon? :)
The long-term fix would be to predict recursively, that is already done in the follow-up time series PR